collect_app/src/main/java/org/odk/collect/android/material/MaterialFullScreenDialogFragment.java

```diff
@@ -0,0 +1,82 @@
+package org.odk.collect.android.material;
+
+import android.app.Dialog;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.Toolbar;
+import androidx.fragment.app.DialogFragment;
+
+import org.odk.collect.android.R;
+
+/**
+ * Provides an implementation of Material's "Full Screen Dialog"
+ * (https://material.io/components/dialogs/#full-screen-dialog) as no implementation currently
+ * exists in the Material Components framework
+ */
+public class MaterialFullScreenDialogFragment extends DialogFragment {
```

Comment:
Could be abstract right? since it's just used to create other classes.

Reply:
Yeah totally. Will add that.